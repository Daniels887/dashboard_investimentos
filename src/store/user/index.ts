export { default as userReducer } from './reducer'
export * from './types'
