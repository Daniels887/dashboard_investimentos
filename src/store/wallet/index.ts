export { default as walletReducer } from './reducer'
export * from './types'
