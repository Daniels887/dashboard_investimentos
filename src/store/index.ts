import { combineReducers, createStore, CombinedState } from 'redux'
import { userReducer } from './user'
import { walletReducer } from './wallet'

import {
  useDispatch as useReduxDispatch,
  useSelector as useReduxSelector,
  TypedUseSelectorHook
} from 'react-redux'

const appReducer = combineReducers({
  user: userReducer,
  wallet: walletReducer
})

export const rootReducer: typeof appReducer = (state, action) => {
  let newState = {
    ...state
  } as CombinedState<RootState> | undefined

  return appReducer(newState, action)
}

type RootState = ReturnType<typeof appReducer>

export const useSelector: TypedUseSelectorHook<RootState> = useReduxSelector

type ExtractedActions = typeof appReducer extends (
  state: CombinedState<RootState>,
  action: infer T
) => RootState
  ? T
  : never

type ActionTypes = ExtractedActions

type RootDispatch = <T extends ActionTypes>(action: T) => T

export const useDispatch = (): RootDispatch => useReduxDispatch()

const store = createStore(rootReducer)

export { store }
