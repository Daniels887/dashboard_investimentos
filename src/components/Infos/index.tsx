import React from 'react'
import { Box, Typography, Grid } from '@material-ui/core'
import { useSelector } from '../../store'

interface IProps {
  showValues: boolean
}

const Infos: React.FC<IProps> = ({ showValues }) => {
  const user = useSelector((state) => state.user)

  return (
    <Box m={2} ml={{ xs: 0, md: 4 }} textAlign={{ xs: 'center', md: 'left' }}>
      <Grid container>
        <Grid item xs={12} md={2}>
          <Box display="flex" flexDirection="column" mt={{ xs: 2, md: 0 }}>
            <Typography style={{ fontWeight: 'bold' }}>
              Saldo em conta corrente
            </Typography>
            <Box color="grey.600" mt={1}>
              {showValues ? (
                <Typography variant="h5">R$ {user.sale}</Typography>
              ) : (
                <Typography variant="h5">
                  R$ &#9679; &#9679; &#9679; &#9679; &#9679;
                </Typography>
              )}
            </Box>
          </Box>
        </Grid>
        <Grid item xs={12} md={2}>
          <Box display="flex" flexDirection="column" mt={{ xs: 2, md: 0 }}>
            <Typography style={{ fontWeight: 'bold' }}>
              Disponível para investir
            </Typography>
            <Box color="grey.600" mt={1}>
              {showValues ? (
                <Typography variant="h5">R$ {user.available}</Typography>
              ) : (
                <Typography variant="h5">
                  R$ &#9679; &#9679; &#9679; &#9679; &#9679;
                </Typography>
              )}
            </Box>
          </Box>
        </Grid>
        <Grid item xs={12} md={2}>
          <Box display="flex" flexDirection="column" mt={{ xs: 2, md: 0 }}>
            <Typography style={{ fontWeight: 'bold' }}>
              Patrimônio total:
            </Typography>
            <Box color="grey.600" mt={1}>
              {showValues ? (
                <Typography variant="h5">R$ {user.patrimony}</Typography>
              ) : (
                <Typography variant="h5">
                  R$ &#9679; &#9679; &#9679; &#9679; &#9679;
                </Typography>
              )}
            </Box>
          </Box>
        </Grid>
      </Grid>
    </Box>
  )
}

export default Infos
