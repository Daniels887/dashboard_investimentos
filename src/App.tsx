import React, { useState } from 'react'
import { hot } from 'react-hot-loader/root'
import theme from './styles/theme'
import { CssBaseline, MuiThemeProvider } from '@material-ui/core'
import { Grid, Box } from '@material-ui/core'
import MenuDashboard from './components/Menu'
import Header from './components/Header'
import Infos from './components/Infos'
import TabsDashboard from './components/Tabs'
import { Provider } from 'react-redux'
import { store } from './store'

const App: React.FC = () => {
  const [showValues, setShowValues] = useState(true)

  return (
    <>
      <MuiThemeProvider theme={theme}>
        <CssBaseline />
        <Provider store={store}>
          <Grid container>
            <Grid item xs={12} md={2} lg={1}>
              <MenuDashboard />
            </Grid>
            <Grid item xs={12} md={10} lg={11}>
              <Box>
                <Header setShowValues={setShowValues} showValues={showValues} />
                <Infos showValues={showValues} />
                <TabsDashboard showValues={showValues} />
              </Box>
            </Grid>
          </Grid>
        </Provider>
      </MuiThemeProvider>
    </>
  )
}

export default hot(App)
