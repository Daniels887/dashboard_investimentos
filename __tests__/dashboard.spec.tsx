import React from 'react'
import renderer from 'react-test-renderer'
import Infos from '../src/components/Infos'
import { store } from '../src/store'
import { Provider } from 'react-redux'

it('Infos renders correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Infos showValues={true} />
      </Provider>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
