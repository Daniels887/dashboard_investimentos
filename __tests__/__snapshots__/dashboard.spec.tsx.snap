// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Infos renders correctly 1`] = `
<div
  className="MuiBox-root MuiBox-root-1"
>
  <div
    className="MuiGrid-root MuiGrid-container"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-2"
    >
      <div
        className="MuiBox-root MuiBox-root-2"
      >
        <p
          className="MuiTypography-root MuiTypography-body1"
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        >
          Saldo em conta corrente
        </p>
        <div
          className="MuiBox-root MuiBox-root-3"
          id="sale-hide"
        >
          <h5
            className="MuiTypography-root MuiTypography-h5"
          >
            R$ 
            5.620.080,85
          </h5>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-2"
    >
      <div
        className="MuiBox-root MuiBox-root-4"
      >
        <p
          className="MuiTypography-root MuiTypography-body1"
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        >
          Disponível para investir
        </p>
        <div
          className="MuiBox-root MuiBox-root-5"
          id="available-hide"
        >
          <h5
            className="MuiTypography-root MuiTypography-h5"
          >
            R$ 
            10.000.000,00
          </h5>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-2"
    >
      <div
        className="MuiBox-root MuiBox-root-6"
      >
        <p
          className="MuiTypography-root MuiTypography-body1"
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        >
          Patrimônio total:
        </p>
        <div
          className="MuiBox-root MuiBox-root-7"
          id="patrimony-hide"
        >
          <h5
            className="MuiTypography-root MuiTypography-h5"
          >
            R$ 
            10.000.000,00
          </h5>
        </div>
      </div>
    </div>
  </div>
</div>
`;
